### ISSUE TITLE
Problem with build the application after upgrade

### ISSUE THREAD

**Issue by wiktoriasobczyk** [flutter/flutter#47622]
After upgrade  I have this problem with compiling the App. I've tried ```flutter clean``` etc and I have still the same problem. 

**Exception**
```
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/graphql_flutter-2.1.0/lib/src/widgets/cache_provider.dart:64:30: Error: Getter not found: 'suspending'.
      case AppLifecycleState.suspending:
                             ^^^^^^^^^^
Target kernel_snapshot failed: Exception: Errors during snapshot creation: null
build failed.

FAILURE: Build failed with an exception.

* Where:
Script 'C:\flutter\packages\flutter_tools\gradle\flutter.gradle' line: 780

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDebug'.
> Process 'command 'C:\flutter\bin\flutter.bat'' finished with non-zero exit value 1
```

**Command: flutter doctor -v **
``` 
 • Flutter version 1.12.13+hotfix.5 at C:\flutter
    • Framework revision 27321ebbad (12 days ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0


[√] Android toolchain - develop for Android devices (Android SDK version 29.0.0)
    • Android SDK at C:\SDK
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.0
    • ANDROID_HOME = C:\SDK
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.
[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[!] IntelliJ IDEA Ultimate Edition (version 2019.2)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA 2018.3.5
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.dev/intellij-setup/#installing-the-plugins

[√] VS Code (version 1.41.1)
    • VS Code at C:\Users\wikto\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.7.1

[√] Connected device (1 available)
    • ANE LX1 • 9WVDU18324004145 • android-arm64 • Android 9 (API 28
```
**Command: ```flutter run --verbose ``` **
```
[  +24 ms] executing: [C:\flutter\] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +65 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 27321ebbad34b0a3fafe99fac037102196d655ff
[   +1 ms] executing: [C:\flutter\] git describe --match v*.*.* --first-parent --long --tags
[  +56 ms] Exit code 0 from: git describe --match v*.*.* --first-parent --long --tags
[   +1 ms] v1.12.13+hotfix.5-0-g27321ebba
[   +9 ms] executing: [C:\flutter\] git rev-parse --abbrev-ref --symbolic @{u}
[  +53 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/stable
[        ] executing: [C:\flutter\] git ls-remote --get-url origin
[  +32 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +77 ms] executing: [C:\flutter\] git rev-parse --abbrev-ref HEAD
[  +36 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[ +105 ms] executing: C:\SDK\platform-tools\adb.exe devices -l
[  +21 ms] Exit code 0 from: C:\SDK\platform-tools\adb.exe devices -l
[        ] List of devices attached
           9WVDU18324004145       device product:ANE-LX1 model:ANE_LX1 device:HWANE transport_id:1
[  +14 ms] C:\SDK\platform-tools\adb.exe -s 9WVDU18324004145 shell getprop
[ +114 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[  +10 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +6 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +82 ms] Found plugin path_provider at C:\flutter\.pub-cache\hosted\pub.dartlang.org\path_provider-1.5.1\
[ +198 ms] Found plugin path_provider at C:\flutter\.pub-cache\hosted\pub.dartlang.org\path_provider-1.5.1\
[  +71 ms] Generating C:\inz\timetable-app\timetable_app\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java
[  +50 ms] ro.hardware = hi6250
[   +1 ms] ro.build.characteristics = default
[  +49 ms] Launching lib\main.dart on ANE LX1 in debug mode...
[  +14 ms] executing: C:\SDK\platform-tools\adb.exe -s 9WVDU18324004145 shell -x logcat -v time -s flutter
[   +6 ms] executing: C:\SDK\platform-tools\adb.exe version
[  +35 ms] Android Debug Bridge version 1.0.41
           Version 29.0.1-5644136
           Installed as C:\SDK\platform-tools\adb.exe
[   +2 ms] executing: C:\SDK\platform-tools\adb.exe start-server
[  +22 ms] Building APK
[  +21 ms] Running Gradle task 'assembleDebug'...
[   +1 ms] gradle.properties already sets `android.enableR8`
[   +4 ms] Using gradle from C:\inz\timetable-app\timetable_app\android\gradlew.bat.
[  +12 ms] executing: C:\Program Files\Android\Android Studio\jre\bin\java -version
[  +96 ms] Exit code 0 from: C:\Program Files\Android\Android Studio\jre\bin\java -version
[        ] openjdk version "1.8.0_202-release"
           OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
           OpenJDK 64-Bit Server VM (build 25.202-b03, mixed mode)
[   +4 ms] executing: [C:\inz\timetable-app\timetable_app\android\] C:\inz\timetable-app\timetable_app\android\gradlew.bat
-Pverbose=true -Ptarget=C:\inz\timetable-app\timetable_app\lib\main.dart -Ptrack-widget-creation=true
-Pfilesystem-scheme=org-dartlang-root -Ptarget-platform=android-arm64 assembleDebug
[+3917 ms] > Task :app:compileFlutterBuildDebug
[   +2 ms] [  +23 ms] executing: [C:\flutter\] git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +1 ms] [  +62 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +1 ms] [        ] 27321ebbad34b0a3fafe99fac037102196d655ff
[   +1 ms] [        ] executing: [C:\flutter\] git describe --match v*.*.* --first-parent --long --tags
[   +4 ms] [  +40 ms] Exit code 0 from: git describe --match v*.*.* --first-parent --long --tags
[   +3 ms] [        ] v1.12.13+hotfix.5-0-g27321ebba
[        ] [   +7 ms] executing: [C:\flutter\] git rev-parse --abbrev-ref --symbolic @{u}
[   +1 ms] [  +39 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[   +1 ms] [        ] origin/stable
[   +1 ms] [        ] executing: [C:\flutter\] git ls-remote --get-url origin
[        ] [  +33 ms] Exit code 0 from: git ls-remote --get-url origin
[   +1 ms] [        ] https://github.com/flutter/flutter.git
[        ] [  +80 ms] executing: [C:\flutter\] git rev-parse --abbrev-ref HEAD
[        ] [  +30 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] [        ] stable
[   +2 ms] [  +18 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +1 ms] [   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[  +30 ms] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[  +18 ms] [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[   +3 ms] [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[   +6 ms] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[   +5 ms] [  +81 ms] Initializing file store
[        ] [  +21 ms] kernel_snapshot: Starting due to {}
[        ] [  +13 ms] C:\flutter\bin\cache\dart-sdk\bin\dart.exe
C:\flutter\bin\cache\artifacts\engine\windows-x64\frontend_server.dart.snapshot --sdk-root
C:\flutter\bin\cache\artifacts\engine\common\flutter_patched_sdk/ --target=flutter -Ddart.developer.causal_async_stacks=true
-Ddart.vm.profile=false -Ddart.vm.product=false
--bytecode-options=source-positions,local-var-info,debugger-stops,instance-field-initializers,keep-unreachable-code,avoid-closure-call-instructions --enable-asserts --track-widget-creation --no-link-platform --packages C:\inz\timetable-app\timetable_app\.packages
--output-dill C:\inz\timetable-app\timetable_app\.dart_tool\flutter_build\3b6d86f7deb896db22771da9b8bcd385\app.dill --depfile
C:\inz\timetable-app\timetable_app\.dart_tool\flutter_build\3b6d86f7deb896db22771da9b8bcd385\kernel_snapshot.d
package:timetable_app/main.dart
[+7277 ms] [+8346 ms] 
[   +4 ms]            Compiler message:
[   +1 ms] [   +4 ms] /C:/flutter/.pub-cache/hosted/pub.dartlang.org/graphql_flutter-2.1.0/lib/src/widgets/cache_provider.dart:64:30:
Error: Getter not found: 'suspending'.
[   +3 ms] [        ]       case AppLifecycleState.suspending:
[   +1 ms] [        ]                              ^^^^^^^^^^
[+4093 ms] [+4059 ms] Persisting file store
[  +17 ms] [  +11 ms] Done persisting file store
[   +8 ms] [  +12 ms] "flutter assemble" took 12 600ms.
[  +12 ms] [   +1 ms] Target kernel_snapshot failed: Exception: Errors during snapshot creation: null
[   +2 ms] build failed.
[   +1 ms] #0      throwToolExit (package:flutter_tools/src/base/common.dart:28:3)
[  +18 ms] #1      AssembleCommand.runCommand (package:flutter_tools/src/commands/assemble.dart:183:7)
[   +1 ms] <asynchronous suspension>
[        ] #2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:615:18)
[   +1 ms] <asynchronous suspension>
[        ] #3      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:515:33)
[   +1 ms] <asynchronous suspension>
[        ] #4      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:146:29)
[   +2 ms] #5      _rootRun (dart:async/zone.dart:1126:13)
[  +15 ms] #6      _CustomZone.run (dart:async/zone.dart:1023:19)
[  +13 ms] #7      _runZoned (dart:async/zone.dart:1518:10)
[   +4 ms] #8      runZoned (dart:async/zone.dart:1465:12)
[   +1 ms] #9      AppContext.run (package:flutter_tools/src/base/context.dart:145:18)
[   +1 ms] #10     FlutterCommand.run (package:flutter_tools/src/runner/flutter_command.dart:505:20)
[   +1 ms] #11     CommandRunner.runCommand (package:args/command_runner.dart:197:27)
[   +2 ms] #12     FlutterCommandRunner.runCommand.<anonymous closure>
(package:flutter_tools/src/runner/flutter_command_runner.dart:413:21)
[   +1 ms] <asynchronous suspension>
[   +1 ms] #13     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:146:29)
[   +2 ms] #14     _rootRun (dart:async/zone.dart:1126:13)
[   +1 ms] #15     _CustomZone.run (dart:async/zone.dart:1023:19)
[   +1 ms] #16     _runZoned (dart:async/zone.dart:1518:10)
[        ] #17     runZoned (dart:async/zone.dart:1465:12)
[        ] #18     AppContext.run (package:flutter_tools/src/base/context.dart:145:18)
[   +1 ms] #19     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:364:19)
[   +1 ms] #20     CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:112:25)
[   +1 ms] #21     new Future.sync (dart:async/future.dart:224:31)
[   +2 ms] #22     CommandRunner.run (package:args/command_runner.dart:112:14)
[        ] #23     FlutterCommandRunner.run (package:flutter_tools/src/runner/flutter_command_runner.dart:250:18)
[        ] #24     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:63:22)
[        ] #25     _rootRun (dart:async/zone.dart:1126:13)
[        ] #26     _CustomZone.run (dart:async/zone.dart:1023:19)
[   +1 ms] #27     _runZoned (dart:async/zone.dart:1518:10)
[  +16 ms] #28     runZoned (dart:async/zone.dart:1502:12)
[   +4 ms] #29     run.<anonymous closure> (package:flutter_tools/runner.dart:61:18)
[   +1 ms] <asynchronous suspension>
[        ] #30     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:146:29)
[   +3 ms] #31     _rootRun (dart:async/zone.dart:1126:13)
[   +3 ms] #32     _CustomZone.run (dart:async/zone.dart:1023:19)
[   +1 ms] #33     _runZoned (dart:async/zone.dart:1518:10)
[        ] #34     runZoned (dart:async/zone.dart:1465:12)
[   +1 ms] #35     AppContext.run (package:flutter_tools/src/base/context.dart:145:18)
[        ] #36     runInContext (package:flutter_tools/src/context_runner.dart:64:24)
[   +1 ms] #37     run (package:flutter_tools/runner.dart:50:10)
[   +1 ms] #38     main (package:flutter_tools/executable.dart:65:9)
[   +1 ms] #39     main (file:///C:/flutter/packages/flutter_tools/bin/flutter_tools.dart:8:3)
[   +1 ms] #40     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:305:32)
[   +1 ms] #41     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:174:12)
[   +8 ms] FAILURE: Build failed with an exception.
[   +1 ms] * Where:
[   +1 ms] Script 'C:\flutter\packages\flutter_tools\gradle\flutter.gradle' line: 780
[   +1 ms] * What went wrong:
[        ] Execution failed for task ':app:compileFlutterBuildDebug'.
[   +1 ms] > Process 'command 'C:\flutter\bin\flutter.bat'' finished with non-zero exit value 1
[        ] * Try:
[        ] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with
--scan to get full insights.
[   +1 ms] * Get more help at https://help.gradle.org
[   +1 ms] BUILD FAILED in 15s
[   +1 ms] > Task :app:compileFlutterBuildDebug FAILED
[        ] 1 actionable task: 1 executed
[ +511 ms] Running Gradle task 'assembleDebug'... (completed in 16,2s)
[   +7 ms] "flutter run" took 17 112ms.
Gradle task assembleDebug failed with exit code 1

#0      throwToolExit (package:flutter_tools/src/base/common.dart:28:3)
#1      buildGradleApp (package:flutter_tools/src/android/gradle.dart:387:7)
#2      _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:73:64)
#3      _rootRunUnary (dart:async/zone.dart:1134:38)
#4      _CustomZone.runUnary (dart:async/zone.dart:1031:19)
#5      _FutureListener.handleValue (dart:async/future_impl.dart:139:18)
#6      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:680:45)
#7      Future._propagateToListeners (dart:async/future_impl.dart:709:32)
#8      Future._completeWithValue (dart:async/future_impl.dart:524:5)
#9      _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:32:15)
#10     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:290:13)
#11     _DefaultProcessUtils.stream (package:flutter_tools/src/base/process.dart)
#12     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:73:64)
#13     _rootRunUnary (dart:async/zone.dart:1134:38)
#14     _CustomZone.runUnary (dart:async/zone.dart:1031:19)
#15     _FutureListener.handleValue (dart:async/future_impl.dart:139:18)
#16     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:680:45)
#17     Future._propagateToListeners (dart:async/future_impl.dart:709:32)
#18     Future._completeWithValue (dart:async/future_impl.dart:524:5)
#19     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:554:7)
#20     _rootRun (dart:async/zone.dart:1126:13)
#21     _CustomZone.run (dart:async/zone.dart:1023:19)
#22     _CustomZone.runGuarded (dart:async/zone.dart:925:7)
#23     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:965:23)
#24     _microtaskLoop (dart:async/schedule_microtask.dart:43:21)
#25     _startMicrotaskLoop (dart:async/schedule_microtask.dart:52:5)
#26     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
#27     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:175:5)
```

**Comment by kpguru20001** [comment]
Hey @wiktoriasobczyk were you able to solve the issue

**Comment by wiktoriasobczyk** [comment]
Hey @kpguru20001 I was a problem with Graphql library. I've removed  lib and  it solved this problem

**Comment by Xgamefactory** [comment]
how sad that gradle still not able to reproduce normal error message to identify issue.

**Comment by kpguru20001** [comment]
But @wiktoriasobczyk what did  you use alternative to the flutter_graphql package

**Comment by jacky0707** [comment]
I met the same issue here, but flutter_graphql is the only way I get the data from backend...

**Comment by kpguru20001** [comment]
@wiktoriasobczyk @jacky0707 so I was able to figure out a way to do it with flutter_graphql
so after asking the maintainers of the plugin, I was able to run the code using 

graphql_flutter: ^2.1.1-beta.5 
Just change the dependency in the pubspec

Also, keep in mind the team will be releasing 3.0.0 which is in beta and 2.1.1 is not gonna be released as they have clubbed 2.1.1 to 3.0.0 so when release you can update your code with new readme as there is no readme or examples of the beta of 3.0.0 ( SO recommend using 2.1.1) which is pretty similar to 2.1.0 for which u are getting that issue

Only thing is document will be deprecated use :
`
String getNotifications = """
  query notifications {
    allNotifications {
      message
      date
      id
    }
  }
""";

documentNode: gql(getNotifications),
`

**Comment by jacky0707** [comment]
@kpguru20001  thx a lot!
